src/server/api/routers/user.ts

```diff
@@ -46,7 +88,46 @@ export const userRouter = createTRPCRouter({
         },
       });
     }),
+  deleteById: protectedProcedure
+    .input(z.object({ id: z.string() }))
+    .mutation(async ({ input }) => {
+      try {
+        return await db.$transaction(async (prisma) => {
+          await prisma.profile.deleteMany({ where: { userId: input.id } });
+          await prisma.account.deleteMany({ where: { userId: input.id } });
+          await prisma.session.deleteMany({ where: { userId: input.id } });
+          return await prisma.user.delete({ where: { id: input.id } });
+        });
+      } catch (error) {
+        console.error(error);
+        throw new TRPCError({
+          code: "BAD_REQUEST",
+          message: `User ${input.id} not found in database`,
+        });
+      }
+    }),
 
+  deleteByIds: protectedProcedure
+    .input(z.object({ ids: z.array(z.string()) }))
+    .mutation(async ({ input }) => {
+      try {
+        return await db.$transaction(async (prisma) => {
+          for (const id of input.ids) {
+            await prisma.profile.deleteMany({ where: { userId: id } });
+            await prisma.account.deleteMany({ where: { userId: id } });
+            await prisma.session.deleteMany({ where: { userId: id } });
+            await prisma.user.delete({ where: { id } });
```

Comment:
Can we cascade instead? Don't feel like we need to do it this way although nice job putting it into a transaction

Reply:
If prisma supports soft delete we can look into that too to avoid accidents